message PlayerMessage {

	required int32 playerId = 0;
	optional string name = 1;
	required int32 health = 2;
	required InventoryMessage inventory = 3;
	required LocationMessage location = 4;
	required PathMessage path = 5;
	
}

message LocationMessage {

	required int32 x = 0;
	required int32 y = 0;

}

message ItemMessage {

	required string itemName;

}

message MapMessage {

	repeated RoomMessage room = 0;

}

message RoomMessage {

	required LocationMessage location = 0;
	required AdjacencyListMessage adjacencyList = 1;
	required InventoryMessage inventory = 2;
	required int32 type = 3;
	optional string puzzle = 4;
	optional string solution = 5;
	required PlayerListMessage playerList = 6;

}

message AdjacencyListMessage {

	required int32 north = 0;
	required int32 east = 1;
	required int32 south = 2;
	required int32 west = 3;

}

message PlayerListMessage {

	repeated PlayerMessage player = 0;

}

message InventoryMessage {

	repeated ItemMessage item = 0;

}

message PathMessage {

	repeated LocationMessage location = 0;

}

message GameMessage {

	required PlayerListMessage playerList = 0;
	required MapMessage map = 1;

}
